src/BalanceLaws/prog_prim_conversion.jl

```diff
@@ -0,0 +1,43 @@
+# Add `Primitive` type to BalanceLaws, AtmosModel
+
+# Vars wrapper
+function prognostic_to_primitive!(
+    prim::AbstractArray,
+    prog::AbstractArray,
+    bl::BalanceLaw,
+    aux::AbstractArray,
+)
+    FT = eltype(prim)
+    prognostic_to_primitive!(
+        Vars{vars_state(bl, Primitive(), FT)}(prim),
+        Vars{vars_state(bl, Prognostic(), FT)}(prog),
+        bl,
+        Vars{vars_state(bl, Auxiliary(), FT)}(aux),
+    )
+end
+function primitive_to_prognostic!(
+    prog::AbstractArray,
+    prim::AbstractArray,
+    bl::BalanceLaw,
+    aux::AbstractArray,
+)
+    FT = eltype(prog)
+    primitive_to_prognostic!(
+        Vars{vars_state(bl, Prognostic(), FT)}(prog),
+        Vars{vars_state(bl, Primitive(), FT)}(prim),
+        bl,
+        Vars{vars_state(bl, Auxiliary(), FT)}(aux),
+    )
+end
```

Comment:
I like this! Thanks for adding it

Reply:
I haven't yet added a test that calls the array interface. I'll try to add that when I do last-minute cleaning on this PR.